### PR TITLE
Routing fix for simple-auth example

### LINKED ISSUE #8783: Simple Auth Example - Application Routing Failure
## Description

When running the `simple-auth` example, under certain situations the user can end up on the `/app/login` url but with nothing being rendered on-screen.

### Steps to reproduce

When running the `simple-auth` example, if:

* A user navigates to the page
* Goes through the login flow
* Gets redirected to `/app/profile`
* Manually removes the `user` from `localStorage`
* Refreshes the browser
* The user will end up on the `/app/login` url but with nothing being rendered

Alternatively, before even logging-in, if the user navigates straight to the `/app/profile` route, they will experience the blank screen.

### Expected result

They should see the `<Login>` component render.

### Actual result

Nothing renders, though the URL is updated in the address bar.

### Environment

```
  System:
    OS: macOS 10.14
    CPU: x64 Intel(R) Core(TM) i5-7267U CPU @ 3.10GHz
    Shell: 5.3 - /bin/zsh
  Binaries:
    Node: 10.9.0 - ~/.nvm/versions/node/v10.9.0/bin/node
    Yarn: 1.10.1 - /usr/local/bin/yarn
    npm: 6.2.0 - ~/.nvm/versions/node/v10.9.0/bin/npm
  Browsers:
    Chrome: 69.0.3497.100
    Firefox: 62.0.2
    Safari: 12.0
  npmPackages:
    gatsby: ^2.0.0 => 2.0.18 
    gatsby-plugin-react-helmet: ^3.0.0 => 3.0.0 
```

### PR DESCRIPTION
**Pull request by Siyfion**
Switch from `navigate` to `redirectTo`, fixes #8783.
Not 100% sure _why_ the old code didn't work, but this does fix it.

### CODE REVIEW

File: examples/simple-auth/src/components/PrivateRoute.js [hunk 1]
```diff
@@ -1,12 +1,12 @@
 import React from "react"
 import PropTypes from "prop-types"
-import { navigate } from "@reach/router"
+import { redirectTo } from "@reach/router"
```
Comment (DSchau) [inline examples/simple-auth/src/components/PrivateRoute.js]: Would you be able to try [this approach](https://www.gatsbyjs.org/docs/gatsby-link/#programmatic-navigation)?

That's our recommended approach! (Not sure why it wasn't using that previously 😅 )

Comment (Siyfion) [reply examples/simple-auth/src/components/PrivateRoute.js]: Yes, of course, I'll do that as soon as I'm off the 🚋

Comment (Siyfion) [reply examples/simple-auth/src/components/PrivateRoute.js]: @DSchau should I change the other reference to `redirectTo` as well? (In the Login component?)

Comment (DSchau) [reply examples/simple-auth/src/components/PrivateRoute.js]: @Siyfion yes please :)

Comment (Siyfion) [reply examples/simple-auth/src/components/PrivateRoute.js]: Yup, all working fine.

**Pull request merged**